overlord/ifacestate/helpers.go

```diff
@@ -375,28 +399,33 @@ ConnsLoop:
 		var updateStaticAttrs bool
 		staticPlugAttrs := connState.StaticPlugAttrs
 		staticSlotAttrs := connState.StaticSlotAttrs
-
-		// XXX: Refresh the copy of the static connection attributes for "content"
-		// and "system-files" interfaces.
-		// This is a partial and temporary solution to https://bugs.launchpad.net/snapd/+bug/1825883
-		// and https://bugs.launchpad.net/snapd/+bug/1942266.
-		switch plugInfo.Interface {
-		case "content":
-			var plugContent, slotContent string
-			plugInfo.Attr("content", &plugContent)
-			slotInfo.Attr("content", &slotContent)
-
-			if plugContent != "" && plugContent == slotContent {
-				staticPlugAttrs = utils.NormalizeInterfaceAttributes(plugInfo.Attrs).(map[string]interface{})
-				staticSlotAttrs = utils.NormalizeInterfaceAttributes(slotInfo.Attrs).(map[string]interface{})
-				updateStaticAttrs = true
+		newStaticPlugAttrs := utils.NormalizeInterfaceAttributes(plugInfo.Attrs).(map[string]interface{})
+		newStaticSlotAttrs := utils.NormalizeInterfaceAttributes(slotInfo.Attrs).(map[string]interface{})
+
+		if canReload {
+			// if the interface was originally autoconnected, update the static attrs if it would
+			// still be allowed to autoconnect. Otherwise, update the static attrs if it would still
+			// be allowed to regular connect.
+			if connState.Auto && !connState.ByGadget {
```

Comment:
why the !connState.ByGadget part here?

Reply:
My understanding is that connections that correspond to `connState.ByGadget == true`, `connState.Auto == true`  are those that were originally made *not* by an `allow-auto-connection` rule in the snap's decl assert, but by a connection directive in the gadget in conjuction with an `allow-connection` rule in the snap's decl assert. Because these connections were initially made per the `allow-connection` rule, they should be updated per the `allow-connection` rule.

It is possible, if not even likely that these connections would be denied by the `allow-auto-connection` rule (hence why they were connected using a gadget directive) and therefore they would never properly update their static attrs

Reply:
I see, we have the same check in doConnect